### PR TITLE
Ported to python3

### PR DESCRIPTION
**Pull request by 8032**
* Refactored the library from python 2 to python 3.
* Removed custom URLlib
* Added additional method to `YouTrackIssue` to return a dictionary of the values
* Added `README.rst` to enable building module

Feel free to update and amend as necessary

### CODE REVIEW

**Comment by acorbin3** [comment]
We should really consider upgrading to python 3 since 2.7 is end of life Jan 1 2020

**Comment by stappersg** [comment]
Cool. I could succesfull do
```python3
myissue = yt.getIssue(issue_name)
print(myissue)
```

In other words:  My test with the Python3 port was succesfull and I happy with it    :-)


Groeten Geert

P.S.
`README.rst` seems to be empty ...

**Comment by 8032** [comment]
I have updated the packaging scripts and added classifiers for PIP to use github flavoured markdown instead of `rst` file.

**Comment by ikrauchanka** [comment]
when?

**Comment by 8032** [comment]
Any update on when this PR can be merged or reviewed further?

**Comment by allvo** [comment]
@ikrauchanka I'm sorry for not processing your PR earlier. Have created a review, after the points are addressed, we will test the PR and merge it.

**Comment by rafal-jaworski** [comment]
@allvo any updates? :>

**Comment by 8032** [comment]
@allvo please advise if there are any remaining items to be updated?

File: youtrack/connection.py [hunk 1]
```diff
@@ -568,8 +566,8 @@ def createGroup(self, group):
         return content
 
     def addUserRoleToGroup(self, group, userRole):
-        url_group_name = urllib.parse.quote(utf8encode(group.name))
-        url_role_name = urllib.parse.quote(utf8encode(userRole.name))
+        url_group_name = urllib.parse.quote(group.name)
```
[the remark this answers is not in the document]
Comment (8032) [reply youtrack/connection.py]: Using python 3 is unicode native so it wasn't necessary.

**Comment by 8032** [comment]
Any update on this PR being merged?

**Comment by francisg-gc** [comment]
@allvo looks like everything is ok, could this be merged finally please?

**Comment by stappersg** [comment]
I wonder what is blocking the merge of this pull request.

And if [JetBrains](https://www.jetbrains.com/) has no intention to actual merge this PR, it would be polite of then to say so.  A simple (and blunt) pressing the `close pull request`  button would end waiting of **JetBrains** approval for going to Python3 with this library.

> Keep Evolving

**Pull request merged**